samples/azure/azure-service-fabric-routing/Core_6/CandidateVoteCount/EndpointCommunicationListener.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Fabric;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using NServiceBus;
+
+public class EndpointCommunicationListener :
+    ICommunicationListener
+{
+    StatefulServiceContext context;
+    IEndpointInstance endpointInstance;
+
+    public EndpointCommunicationListener(StatefulServiceContext context)
+    {
+        this.context = context;
+    }
+
+    public async Task<string> OpenAsync(CancellationToken cancellationToken)
+    {
+        Logger.Log = m => ServiceEventSource.Current.ServiceMessage(context, m);
+
+        var partitionInfo = await ServicePartitionQueryHelper.QueryServicePartitions(context.ServiceName, context.PartitionId)
+            .ConfigureAwait(false);
+
+        var endpointConfiguration = new EndpointConfiguration("CandidateVoteCount");
+
+        var transport = endpointConfiguration.ApplyCommonConfiguration();
+
+        ConfigureLocalPartitionsCandidateVoteCount(endpointConfiguration, partitionInfo);
+
+        ConfigureReceiverSideDistributionCandidateVoteCount(transport, partitionInfo);
+
+        ConfigureSenderSideRoutingCandidateVoteCount(transport);
+
+        endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+
+        return null;
+    }
+
+    static void ConfigureLocalPartitionsCandidateVoteCount(EndpointConfiguration endpointConfiguration, PartitionsInformation partitionInfo)
+    {
+        #region ConfigureLocalPartitions-CandidateVoteCount
+
+        endpointConfiguration.RegisterPartitionsForThisEndpoint(
+            localPartitionKey: partitionInfo.LocalPartitionKey,
+            allPartitionKeys: partitionInfo.Partitions);
+
+        #endregion
+    }
+
+    static void ConfigureReceiverSideDistributionCandidateVoteCount(TransportExtensions<AzureServiceBusTransport> transportConfig, PartitionsInformation partitionInfo)
+    {
+        #region ConfigureReceiverSideDistribution-CandidateVoteCount
+
+        var routing = transportConfig.Routing();
+        var receiverSideDistributionConfig = routing.EnableReceiverSideDistribution(partitionInfo.Partitions);
+        receiverSideDistributionConfig.AddPartitionMappingForMessageType<VotePlaced>(
+            mapMessageToPartitionKey: votePlaced =>
+            {
+                return votePlaced.Candidate;
+            });
+
+        #endregion
+    }
+
+    static void ConfigureSenderSideRoutingCandidateVoteCount(TransportExtensions<AzureServiceBusTransport> transportConfig)
+    {
+        #region ConfigureSenderSideRouting-CandidateVoteCount
+
+        var remotePartitions = new[] {"33000", "66000", "99000"};
+
+        Func<TrackZipCode, string> convertStringZipCodeToHighKey = message =>
+        {
+            var zipCodeAsNumber = Convert.ToInt32(message.ZipCode);
```

Comment:
The sharding part is a crucial part of the code. I think we should lead by example.